backend/controllers/productController.ts

```diff
@@ -1,86 +1,55 @@
-import asyncHandler from 'express-async-handler'
-import { Request, Response } from 'express'
-import Product from '../models/productModel'
-
-// @desc  Fetch all products
-// @route GET /api/products
-// @acess Public
-const getProducts = asyncHandler(async (req: any, res: Response) => {
-  const products = await Product.find({})
-  res.json(products)
-})
-
-// @desc  Fetch single product
-// @route GET /api/products/:id
-// @acess Public
-const getProductById = asyncHandler(async (req: any, res: Response) => {
-  const product = await Product.findById(req.params.id)
-
-  if (product) {
-    res.json(product)
-  } else {
-    res.status(404)
-    throw new Error('Product not found')
+import asyncHandler from "express-async-handler";
+import { Response } from "express";
+import Product from "../models/productModel";
+
+export const getProducts = asyncHandler(async (req: any, res: Response) => {
+  const products = await Product.find({});
+  res.json(products);
+});
+
+const findProductById = async (id: string) => {
+  const product = await Product.findById(id);
+  if (!product) {
+    // TODO: use custom error class with status code 404
+    // res.status(404);
```

Comment:
How can I set the response.status if this function doesn't have access to the response object?